feg/gateway/fabfile.py

```diff
@@ -18,48 +18,57 @@
 sys.path.append('../../orc8r')
 import tools.fab.dev_utils as dev_utils  # NOQA
 import tools.fab.types as types
-from fabric.api import local
+from fabric import Connection, task
+from tools.fab.hosts import vagrant_connection
 
 SNOWFLAKE_FEG_FILE = '../../.cache/feg/snowflake'
 NETWORK_ID = 'feg_test'
 FEG_DOCKER_LOCATION = 'docker/'
+FEG_INTEG_DOCKER = "$MAGMA_ROOT/lte/gateway/python/integ_tests/federated_tests/docker"
+AGW_ROOT = '../../lte/gateway'
 
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
 
-def register_feg_gw(location_docker_compose: str = FEG_DOCKER_LOCATION):
+@task
+def register_feg_gw(c, location_docker_compose=FEG_DOCKER_LOCATION):
     """
     Add FEG gateway to orc8r
     Args:
         location_docker_compose: location of docker compose. Default set to
         FEG_DOCKER_LOCATION
     """
     _register_federation_network()
-    _register_feg(location_docker_compose)
+    _register_feg(c, location_docker_compose)
 
 
-def deregister_feg_gw(location_docker_compose: str = FEG_DOCKER_LOCATION):
+@task
+def deregister_feg_gw(c, location_docker_compose=FEG_DOCKER_LOCATION):
     """
     Remove FEG gateway from orc8r and remove certs from FEG gateway
     Args:
         location_docker_compose: location of docker compose. Default set to
         FEG_DOCKER_LOCATION
     """
-    _deregister_feg_gw(location_docker_compose)
-    dev_utils.delete_gateway_certs_from_docker(location_docker_compose)
+    _deregister_feg_gw(c, location_docker_compose)
+    dev_utils.delete_gateway_certs_from_docker(c, location_docker_compose)
 
 
-def check_feg_cloud_connectivity(timeout=5):
+@task
+def check_feg_cloud_connectivity(c, timeout=5):
     """
     Check connectivity of FEG with the cloud using checkin_cli.py
     Args:
         timeout: amount of time the command will retry
     """
-    local("cd docker")
-    local("pwd")
-    dev_utils.local_command_with_repetition(
-        "cd docker; docker compose exec magmad checkin_cli.py", timeout,
-    )
+    with c.cd(AGW_ROOT):
+        c_agw = vagrant_connection(c, 'magma')
```

Comment:
In general: Would `c_gw` be a better name, since it is used for both the AGW and FEG?

Reply:
Good point, this is more general since we use it for both gateways. Done!